File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/common/UaApplicationDescriptor.java
package org.opcfoundation.webserver.types.common;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;

public class UaApplicationDescriptor {
    private final String uri;
    private final LocalizedText name;
    private final String url;

    public UaApplicationDescriptor(
            String uri,
            LocalizedText name,
            String serverUrl)
    {
        this.uri = uri;
        this.name = name;
        this.url = serverUrl + "/" + uri;
    }

    public String getUri() {
        return uri;
    }

    public LocalizedText getName() {
        return name;
    }

    public String getUrl() {
        return url;
    }
}

File: opcua-webservice-node/source/src/server/types/common/UaApplicationDescriptor.ts
import { UaLocalizedText } from "opcua-webapi-ts";

export class UaApplicationDescriptor {
    private readonly _uri: string;
    private readonly _name: UaLocalizedText;
    private readonly _url: string;

    constructor(uri: string, name: UaLocalizedText, serverUrl: string) {
        this._uri = uri;
        this._name = name;
        this._url = serverUrl + "/" + uri;
    }

    get uri(): string {
        return this._uri;
    }

    get name(): UaLocalizedText {
        return this._name;
    }

    get url(): string {
        return this._url;
    }
}